src/test/java/net/datafaker/providers/base/InternetTest.java

```diff
@@ -39,10 +38,20 @@ void emailSubject() {
 
     @Test
     void testUsernameWithSpaces() {
-        final Name name = Mockito.spy(new Name(mockedFaker));
-        doReturn("Compound Name").when(name).firstName();
-        doReturn(name).when(mockedFaker).name();
-        assertThat(mockedFaker.internet().username()).matches("^(\\w+)\\.(\\w+)$");
+        Name name = mock();
+        doReturn("Jin Quan").when(name).firstName();
+        doReturn("D'Artagnan").when(name).lastName();
+
+        BaseFaker mockedFaker = new BaseFaker() {
+            @Override
+            public Name name() {
+                return name;
+            }
+        };
+        assertThat(mockedFaker.internet().username())
+            .doesNotContain(" ", "'")
+            .matches("^(\\w+)\\.(\\w+)$")
+            .matches("^\\p{javaLowerCase}+\\.\\p{javaLowerCase}+$");
```

Comment:
ideally this test should be rewritten

currently it doesn't check much
If I replace implementation of `username` with `return a.a` it will pass